services/apps/src/apps/service/apps/jobs/sharing.clj

```diff
@@ -173,16 +173,28 @@
    (catch ce/clj-http-error? {:keys [body]}
      (str "unable to unshare result folder: " (:error_code (service/parse-json body))))))
 
-;; The apps client isn't used at this time, but it will be once we extend analysis sharing
-;; to HPC apps.
+(defn- unshare-input-file
+  [sharer sharee path]
+  (try+
+   (data-info/unshare-path sharer path sharee)
```

Comment:
The administrative AVU idea does sound interesting, although we'd have to take care to handle the case where multiple analyses that use the same input file are shared with the same user. We'd have to keep a counter in that AVU indicating how many analyses are currently shared with the user. And increment or decrement that count each time, only restoring the original permissions when the counter reaches zero. And we also wouldn't want to reduce permissions on the file if the ownership already has access, which would seem to indicate that the current permission should be used as an indicator. There's a lot of edge cases here.